packages/plugin-core/src/commands/MoveHeader.ts

```diff
@@ -0,0 +1,311 @@
+import {
+  DendronError,
+  NoteProps,
+  NoteUtils,
+  VaultUtils,
+  getSlugger,
+} from "@dendronhq/common-all";
+import {
+  HistoryService,
+  DendronASTDest,
+  DendronASTTypes,
+  AnchorUtils,
+  Processor,
+  RemarkUtils,
+  MDUtilsV5,
+  ProcMode,
+  visit,
+  Heading,
+  Node,
+} from "@dendronhq/engine-server";
+import { LookupControllerV3 } from "../components/lookup/LookupControllerV3";
+import {
+  NoteLookupProvider,
+  NoteLookupProviderSuccessResp,
+} from "../components/lookup/LookupProviderV3";
+import { DENDRON_COMMANDS } from "../constants";
+import { VSCodeUtils } from "../utils";
+import { BasicCommand } from "./base";
+import _ from "lodash";
+import path from "path";
+import { getEngine } from "../workspace";
+import { EngineAPIService } from "../services/EngineAPIService";
+import { delayedUpdateDecorations } from "../features/windowDecorations";
+import { findReferences } from "../utils/md";
+
+type CommandInput =
+  | {
+      dest?: NoteProps;
+    }
+  | undefined;
+type CommandOpts = {
+  origin: NoteProps;
+  nodesToMove: Node[];
+  modifiedOriginTree: Node;
+  originProc: Processor;
+  engine: EngineAPIService;
+} & CommandInput;
+type CommandOutput = {
+  updated: NoteProps[];
+} & CommandOpts;
+
+export class MoveHeaderCommand extends BasicCommand<
+  CommandOpts,
+  CommandOutput
+> {
+  key = DENDRON_COMMANDS.MOVE_HEADER.key;
+
+  headerNotSelectedError = new DendronError({
+    message: "You must first select the header you want to move.",
```

Comment:
Can you add severity minor to these? 

I'm thinking of using severity as a way to determine whether an error gets uploaded to error analytics, as we don't want this kind of error reported.

I'll do a write-up shortly for a error-reporting style guide / sentry docs.

Reply:
Should we create a class such as `UserError` to distinguish between the type of errors (if this is a user error).

Could be still very beneficial to upload user errors to error analytics to have insight into whether some parts of UI are common potholes for users.

Reply:
That's a good point.  At this point, probably better to err on the side of more data than less data- although we have a different analytics avenue that we typically use to understand usage patterns as opposed to the sentry pathway. It may be more appropriate to divert it there.  I'll do some more thinking on this one

Reply:
added severity for now